tests/imageuploadcommand.js

```diff
@@ -78,6 +79,21 @@ describe( 'ImageUploadCommand', () => {
 			expect( getModelData( document ) ).to.equal( `<image src="image.png"></image>[<image uploadId="${ id }"></image>]` );
 		} );
 
+		it( 'should allow to insert image at some custom position', () => {
+			const file = createNativeFileMock();
+			setModelData( document, '[<paragraph>foo</paragraph>]<paragraph>bar</paragraph>' );
+
+			const selectedElement = document.selection.getSelectedElement();
+			const customPosition = ModelPosition.createAt( selectedElement, 2 );
+
+			command.execute( { file, insertAt: customPosition } );
+
+			const id = fileRepository.getLoader( file ).id;
+			expect( getModelData( document ) ).to.equal(
+				`<paragraph>fo</paragraph>[<image uploadId="${ id }"></image>]<paragraph>o</paragraph><paragraph>bar</paragraph>`
```

Comment:
Still `insertAt` is very close to the selection so you don't know if the change really works.